src/ocean/core/SmartUnion.d

```diff
@@ -254,6 +289,33 @@ private template MemberList ( uint i, size_t len, U )
     }
 }
 
+/*******************************************************************************
+
+    Evaluates to a ',' separated list of the names of the members of U, each
+    wrapped with quotes ("").
+
+    Template_Params:
+        i = U member start index
+        U = aggregate to iterate over
+
+    Evaluates to:
+        a ',' separated list of the names of the members of U, each wrapped with
+        quotes ("").
+
+*******************************************************************************/
+
+private template MemberStringList ( uint i, U )
+{
+    static if ( i == U.tupleof.length )
+    {
+        const MemberStringList = "";
+    }
+    else
+    {
+        const MemberStringList = `,"` ~ FieldName!(i, U) ~ `"` ~
+            MemberStringList!(i + 1, U);
+    }
+}
```

Comment:
> Btw. why did you put `MemberStringList` outside the `SmartUnion` and `SmartUnionIntern` struct?

I see why: `MemberList` is outside as well. I didn't read this code for a long time ;)
Still you *could* put `MemberStringList` inside `SmartUnion/Intern`.